libp2p/protocols/pubsub/gossipsub.nim

```diff
@@ -222,12 +223,14 @@ proc heartbeat(g: GossipSub) {.async.} =
       let peers = g.getGossipPeers()
       var sent: seq[Future[void]]
       for peer in peers.keys:
-        sent &= g.peers[peer].send(@[RPCMsg(control: some(peers[peer]))])
+        if peer in g.peers:
+          sent &= g.peers[peer].send(@[RPCMsg(control: some(peers[peer]))])
       checkFutures(await allFinished(sent))
 
       g.mcache.shift() # shift the cache
     except CatchableError as exc:
       trace "exception ocurred in gossipsub heartbeat", exc = exc.msg
+      continue
```

Comment:
Hmm I'm not sure about this one, first of all it won't release the lock (which I removed in another branch), also it will not throttle the errors if there are any errors.. and with Catchable being the exception can be everything.

Reply:
The point is that an error here should not break heartbeat, ie, heartbeat should never exit. The lock _is_ released, because `continue` will exit the `try/except` and `finally` is guaranteed to be executed always.

Reply:
But it would not break even without, it would just execute the sleep, I guessed the intent and changed it btw here:
https://github.com/status-im/nim-libp2p/blob/master/libp2p/protocols/pubsub/gossipsub.nim#L231-L236
I added a little sleep on purpose to avoid errors piling up in terms of CPU/IO.

Reply:
ok, not a bad fix, but I would simply just move the leep into a finally, and just allow it the regular `1.seconds` sleep...

```nim
while true:
  try:
     ...
  except:
     ...
  finally:
    await sleepAsync(1.seonds)
```

Reply:
Well the above you wrote and the previous:

```nim
while true:
  try:
     ...
  except:
     ...
  
  await sleepAsync(1.seonds)
```

Is equivalent if so.